app/controllers/ops_controller/ops_rbac.rb

```diff
@@ -958,13 +958,27 @@ def rbac_group_right_tree(selected)
   def rbac_role_get_details(id)
     @edit = nil
     @record = @role = MiqUserRole.find_by_id(from_cid(id))
-    @role_features = @role.feature_identifiers.sort
-    @features_tree = rbac_build_features_tree
+    @rbac_menu_tree = build_rbac_menu_tree
   end
 
-  def rbac_build_features_tree
+  def build_rbac_menu_tree
     @role = @sb[:typ] == "copy" ? @record.dup : @record if @role.nil? # if on edit screen use @record
-    TreeBuilder.convert_bs_tree(OpsController::RbacTree.build(@role, @role_features, !@edit.nil?)).to_json
+
+    # FIXME: Remove when sandbox isn't stored in session
+    # Make sure tree_state doesn't hold on to old data between requests
+    #
+    unless @sb.fetch_path('trees', 'features_tree').nil?
+      @sb['trees'].delete('features_tree')
```

Comment:
@himdel @skateman I'm guess this isn't normal, but otherwise too much state is carried over between show and edit and back.